libs/core/algorithms/include/hpx/parallel/datapar/iterator_helpers.hpp

```diff
@@ -434,12 +330,6 @@ namespace hpx { namespace parallel { namespace util { namespace detail {
             typename hpx::util::invoke_result<F, V1*, V2*>::type
             callv(F&& f, Iter1& it1, Iter2& it2)
         {
-            if (!is_data_aligned(it1) || !is_data_aligned(it2))
-            {
-                return invoke_vectorized_in2<V1, V2>::call_unaligned(
-                    HPX_FORWARD(F, f), it1, it2);
-            }
```

Comment:
I think its not needed anymore, because from where ever we invoke it (`callv`) we only do it only if both iterators are aligned or else we fallback to `call1`.

Reply:
Sure, I understand. The purpose of an assertion however is to verify preconditions (i.e. properties/values of arguments for a function) without which the function will not do what's expected. Assertions are compiled out in release builds, so they don't impose any performance overheads. They merely ensure that future changes will not violate said preconditions.